File: app/src/main/java/io/github/fvasco/pinpoi/util/ZipGuardInputStream.java
package io.github.fvasco.pinpoi.util;

import java.io.FilterInputStream;
import java.io.IOException;
import java.util.zip.ZipInputStream;

/**
 * Filter input stream to close all, this stream close only current entry;
 *
 * @author Francesco Vasco
 */
public class ZipGuardInputStream extends FilterInputStream {

    public ZipGuardInputStream(ZipInputStream in) {
        super(in);
    }

    @Override
    public void close() throws IOException {
        ((ZipInputStream) this.in).closeEntry();
    }
}

File: app/src/main/java/io/github/fvasco/pinpoi/util/ZipGuardInputStream.kt
package io.github.fvasco.pinpoi.util

import java.io.FilterInputStream
import java.io.IOException
import java.util.zip.ZipInputStream

/**
 * Filter input stream to close all, this stream close only current entry;

 * @author Francesco Vasco
 */
class ZipGuardInputStream(`in`: ZipInputStream) : FilterInputStream(`in`) {

    @Throws(IOException::class)
    override fun close() {
        (this.`in` as ZipInputStream).closeEntry()
    }
}
